Frontend/ui-customer/src/Screens/OnBoardingFlow/EnterMobileNumberScreen/Controller.purs

```diff
@@ -132,16 +141,15 @@ eval (VerifyOTPButtonAction PrimaryButtonController.OnClick) state = do
     updateAndExit state $ GoToAccountSetUp state
 
 eval (MobileNumberEditTextAction (MobileNumberEditorController.TextChanged id value)) state = do
-    let maxLen = mobileNumberMaxLength state.data.countryObj.countryShortCode
     _ <- if length value ==  mobileNumberMaxLength state.data.countryObj.countryShortCode then do
             pure $ hideKeyboardOnNavigation true
             else pure unit
-    let isValidMobileNumber = mobileNumberValidator state.data.countryObj.countryCode state.data.countryObj.countryShortCode value 
-    let newState = state { props = state.props { isValidMobileNumber = isValidMobileNumber
-                                        , btnActiveMobileNumber = if isValidMobileNumber then true else false
+    let validatorResp = mobileNumberValidator state.data.countryObj.countryCode state.data.countryObj.countryShortCode value 
+    let newState = state { props = state.props { isValidMobileNumber = validatorResp == MVR.ValidPrefix || validatorResp == MVR.Valid
+                                        , btnActiveMobileNumber = validatorResp == MVR.Valid 
                                         , countryCodeOptionExpended = false}
-                                        , data = state.data { mobileNumber = if (length value) <= maxLen then value else state.data.mobileNumber}}
-    if  isValidMobileNumber then do 
+                                        , data = state.data { mobileNumber = if validatorResp == MVR.MaxLengthExceeded then state.data.mobileNumber else value}}
+    if  validatorResp == MVR.Valid then do 
```

Comment:
@S-PRAVINKUMAR put this condition inside a variable as it is being used thrice. `validatorResp == MVR.Valid`